### ISSUE TITLE
Is it possible to have a hybrid autoscaling group?

### ISSUE THREAD

**Issue by deinspanjer** [cristim/autospotting#146]
# Github issue #

## Issue type ##

- Feature Idea

## Build number #
446

## Environment ##

- AWS region us-east-1
- Type of environment: VPC

## Summary ##

I have a three node autoscaling group for Docker Swarm managers.  Technically, 3 t2.micros is large enough, but I'd like to have just one on-demand instance and let the others be reasonably priced spots.

Currently, autospotting cannot find any replacement candidates for a t2.micro in the us-east-1 region, and I'm not sure if that is due to price or size or the fact that no such thing as a t2.micro spot exists.

Would it be possible to have one on-demand t2.micro instance and two m3.medium spots managed by autospotting?

**Comment by cristim** [comment]
Sure, that's the main idea, and it works assuming the following:
- AutoSpotting is configured to keep 1 on-demand instance in your group, by setting the `autospotting_on_demand_number` tag to 1 on the group
- the spot price of m3.medium is lower than the hourly on-demand price of t2.micro in the two availability zones where AutoSpotting decided to terminate your on-demand instances. At the moment it doesn't pick the best priced zones, so it's more or less a matter of luck. I'll capture this as an issue, hopefully someone will implement it at some point.

**Comment by deinspanjer** [comment]
Thank you.  I had to do some manual starting and attaching to get things going, but it seems to be working as intended now.

* I used the "run more like this" option to start another manager node but changed the type to a t2.micro.
* When it came up, I attached it to the autoscaling group manually.
* I then let autospotting do its thing and after a while, it ended up with the one on-demand and two spot that I wanted.

I think the biggest trick was the fact that the us-east-1b AZ doesn't appear to have m3.mediums available for spot, so autospotting was having trouble with that.  I worked around this by actually just putting the micro instance in AZ 1b so that the rest of the balancing naturally chose 1a and 1c.
